bin/convert-blast-xml-to-json.py

```diff
@@ -1,23 +1,48 @@
 #!/usr/bin/env python
 
+import argparse
+import bz2
 import sys
-from os.path import basename
 
 from dark.blast.conversion import XMLRecordsReader
 
+
 if __name__ == '__main__':
-    nArgs = len(sys.argv)
-    if nArgs == 1:
-        reader = XMLRecordsReader(sys.stdin)
-        reader.saveAsJSON(sys.stdout)
-    elif nArgs == 2:
-        reader = XMLRecordsReader(sys.argv[1])
-        reader.saveAsJSON(sys.stdout)
-    elif nArgs == 3:
-        reader = XMLRecordsReader(sys.argv[1])
-        with open(sys.argv[2], 'w') as fp:
+
+    parser = argparse.ArgumentParser(
+        description='Convert a BLAST XML file to JSON.',
+        epilog='Give a BLAST XML file and convert it to JSON. Optionally '
+        'compress the JOSN output file.'
+    )
+
+    parser.add_argument(
+        'xml', metavar='BLAST-XML-file', type=str, help='the XML file of '
+        'BLAST output.')
+
+    parser.add_argument(
+        'json', metavar='BLAST-JSON-file', type=str, nargs='?',
+        help='the JSON filename where contents of "xml" should be written to.')
+
+    parser.add_argument(
+        '--compress', default=False, action='store_true',
+        help='If True, compress the json output.')
+
+    args = parser.parse_args()
+
+    if args.compress:
+        if args.json:
+            reader = XMLRecordsReader(args.xml)
+            fp = bz2.BZ2File(args.json, 'w')
             reader.saveAsJSON(fp)
+            fp.close()
+        else:
+            raise ValueError('JSON filename must be specified if output '
```

Comment:
Hmmm... you couldn't find a way to get BZ2File to write to stdout?  Neither could I :-(